elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/metadata/SQLReferenceVisitor.java

```diff
@@ -81,16 +81,25 @@ protected String visitFormulaDimension(ColumnProjection dimension) {
      * @return
      */
     private String visitFormulaColumn(ColumnProjection column) {
-        Queryable source  = column.getSource();
+        return resolveReferences(column.getSource(), column.getExpression(), column.getName());
+    }
 
-        String expr = column.getExpression();
+    /**
+     * Resolve references.
+     *
+     * @param source
+     * @param expr expression with unresolved references
+     * @param logicalName logical column name
+     * @return expression with resolved references
+     */
+    public String resolveReferences(Queryable source, String expr, String logicalName) {
```

Comment:
Should this be private?

Reply:
No, Its called from QueryTranslator's getJoinClause method

Reply:
I'd prefer is this class was kept private to the construction of the SQLReferenceTable.